net/heartbeat.py

```diff
@@ -32,9 +32,9 @@ def __init__(self, only_ip="127.0.0.1"):
         LoopingCall(self._heartbeat).start(10, now=True)
 
     def buildProtocol(self, addr):
-        if self.status in ("starting up", "generating GUID") and self.only_ip != "127.0.0.1":
+        if self.status in ("starting up", "generating GUID") and "127.0.0.1" not in self.only_ip:
```

Comment:
This would be my only worry. I wanted to prevent outside connections during guid generation because it currently sends the username and password over the heartbeat server after guid generation. If they enter any IP in here and don't use SSL it will send that in the clear.